acceptance/status_server_test.go

```diff
@@ -0,0 +1,124 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Bram Gruneir (bram+code@cockroachlabs.com)
+
+// +build acceptance
+
+package acceptance
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/cockroachdb/cockroach/acceptance/localcluster"
+	"github.com/cockroachdb/cockroach/proto"
+	"github.com/cockroachdb/cockroach/util/log"
+)
+
+// get performs an HTTPS GET to the specified path for a specific node.
+func get(t *testing.T, client *http.Client, node *localcluster.Container, path string) []byte {
+	url := fmt.Sprintf("https://%s%s", node.Addr(""), path)
+	resp, err := client.Get(url)
+	if err != nil {
+		t.Fatalf("could not GET %s - %s", url, err)
+	}
+	if resp.StatusCode != http.StatusOK {
+		t.Fatalf("could not GET %s - statuscode: %d", url, resp.StatusCode)
+	}
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		t.Fatalf("could not open body for %s - %s", url, err)
+	}
+	log.Infof("OK response from %s", url)
+	return body
+}
+
+// TestStatusServer starts up an N node cluster and tests the status server on
+// each node.
+func TestStatusServer(t *testing.T) {
+	l := localcluster.Create(*numNodes, stopper)
+	l.ForceLogging = true
+	l.Start()
+	defer l.Stop()
+	checkRangeReplication(t, l, 20*time.Second)
+
+	client := &http.Client{
+		Timeout: 200 * time.Millisecond,
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: true,
+			},
+		},
+	}
+
+	type details struct {
+		NodeID proto.NodeID `json:"nodeID"`
+	}
+
+	// Get the ids for each node
+	idMap := make(map[string]string)
+	for _, node := range l.Nodes {
+		body := get(t, client, node, "/_status/details/local")
+		var detail details
+		if err := json.Unmarshal(body, &detail); err != nil {
+			t.Fatalf("unable to parse details - %s", err)
+		}
+		idMap[node.ID] = detail.NodeID.String()
+	}
+
+	for _, node := range l.Nodes {
+		// Check local responses for the node.
+		body := get(t, client, node, "/_status/details/local")
+		var detail details
+		if err := json.Unmarshal(body, &detail); err != nil {
+			t.Fatalf("unable to parse details - %s", err)
+		}
+		if a, e := detail.NodeID.String(), idMap[node.ID]; a != e {
+			t.Fatalf("local: node ids don't match - expected %s, actual %s", e, a)
+		}
+
+		get(t, client, node, "/_status/gossip/local")
+		get(t, client, node, "/_status/logfiles/local")
+		get(t, client, node, "/_status/logs/local")
+		get(t, client, node, "/_status/stacks/local")
+		get(t, client, node, "/_status/nodes")
+		get(t, client, node, "/_status/nodes/"+idMap[node.ID])
+		get(t, client, node, "/_status/stores")
+
+		// Now perform the same call with internal proxying.
+		for _, otherNode := range l.Nodes {
```

Comment:
Trying all possible pairings of nodes seems excessive; I would just pick one pair of nodes for this part of the test.